subsys/pm/device_runtime.c

```diff
@@ -24,7 +24,9 @@ static void device_pm_callback(const struct device *dev,
 {
 	__ASSERT(retval == 0, "Device set power state failed");
 
-	atomic_set(&dev->pm->state, *state);
+	(void)k_mutex_lock(&dev->pm->lock, K_FOREVER);
+	dev->pm->state = *state;
+	(void)k_mutex_unlock(&dev->pm->lock);
```

Comment:
unlock after broadcast.

Reply:
ok, assuming that this callback will be called by the device in `pm_device_state_set`, the mutex will be locked and not necessary here.